src/libraries/Microsoft.PowerFx.Connectors/Shared/ServiceFunction.cs

```diff
@@ -389,12 +388,28 @@ public async Task<FormulaValue> InvokeAsync(FormulaValue[] args, CancellationTok
             {
                 throw new InvalidOperationException($"Function {Name} can't be invoked.");
             }
-            return await _invoker.InvokeAsync(args, cancel);
+
+            var result = await _invoker.InvokeAsync(args, cancel);
+            ExpressionError er = null;
+
+            if (result is ErrorValue ev && (er = ev.Errors.FirstOrDefault(e => e.Kind == ErrorKind.Network)) != null)
+            {
+                result = FormulaValue.NewError(
+                    new ExpressionError()
+                    {
+                        Kind = er.Kind,
+                        Severity = er.Severity,
+                        Message = $"{_invoker.Namespace.ToDottedSyntax()}.{_invoker.Name} failed: {er.Message}"
+                    },
+                    ev.Type);
+            }
+
```

Comment:
Does PowerApps do this (add the extra error, formated like this)?

For example, this expression:
IfError(Text(MSNWeather.CurrentWeather("", "Imperial").responses.weather.current.temp),
 FirstError.Message)


Returns the string:
MSNWeather.CurrentWeather failed: { "statusCode": 404, "message": "Resource not found" }

Also, this expression returns 1:
IfError(Text(MSNWeather.CurrentWeather("", "Imperial").responses.weather.current.temp),
 CountRows(AllErrors)) // 1